lms/migrations/versions/f6c442c861c4_backfill_assignment_course_id.py

```diff
@@ -0,0 +1,37 @@
+"""Backfill assignment.course_id."""
+
+import sqlalchemy as sa
+from alembic import op
+
+revision = "f6c442c861c4"
+down_revision = "18109b584a5b"
+
+
+def upgrade() -> None:
+    conn = op.get_bind()
+
+    conn.execute(
+        sa.text(
+            """
+        WITH assignment_courses as (
+          SELECT DISTINCT ON (assignment.id) assignment.id as assignment_id, grouping.id  as course_id
```

Comment:
You can play with the [select side on metabase](https://metabase.hypothes.is/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJcblxuXG5cblxuU0VMRUNUIERJU1RJTkNUIE9OIChhc3NpZ25tZW50LmlkKSBhc3NpZ25tZW50LmlkIGFzIGFzc2lnbm1lbnRfaWQsIGdyb3VwaW5nLmlkICBhcyBjb3Vyc2VfaWRcbiAgICAgICAgICBGUk9NIGFzc2lnbm1lbnRcbiAgICAgICAgICBKT0lOIGFzc2lnbm1lbnRfZ3JvdXBpbmcgb24gYXNzaWdubWVudC5pZCA9IGFzc2lnbm1lbnRfZ3JvdXBpbmcuYXNzaWdubWVudF9pZFxuICAgICAgICAgIEpPSU4gZ3JvdXBpbmcgb24gZ3JvdXBpbmcuaWQgPSBncm91cGluZ19pZFxuICAgICAgICAgIC0tIE9ubHkgY291cnNlcywgbm90IHNlY3Rpb25zIG9yIGdyb3Vwc1xuICAgICAgICAgIFdIRVJFIGdyb3VwaW5nLnR5cGUgPSAnY291cnNlJ1xuICAgICAgICAgIC0tIERvbid0IG92ZXJyaWRlIGRhdGEgd2UgYWxyZWFkeSBzZXQgc2luY2Ugd2UgY3JlYXRlZCB0aGUgY29sdW1uXG4gICAgICAgICAgYW5kIGFzc2lnbm1lbnQuY291cnNlX2lkIGlzIG51bGxcbiAgICAgICAgICAtLSBPcmRlciBmb3IgdGhlIGBESVNUSU5DVCBPTmAsIHdoZW4gZHVwbGljYXRlIGZvdW5kIGZvciBvbmUgYXNzaWdubWVudCwgcGljayB0aGUgbGF0ZXN0XG4gICAgICAgICAgT1JERVIgQlkgYXNzaWdubWVudC5pZCwgYXNzaWdubWVudF9ncm91cGluZy5jcmVhdGVkIGRlc2MiLCJ0ZW1wbGF0ZS10YWdzIjp7fX0sImRhdGFiYXNlIjo3fSwiZGlzcGxheSI6InRhYmxlIiwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==)